File: LeetCode/88. Merge Sorted Array/merge.ts
function merge(nums1: number[], m: number, nums2: number[], n: number): number[] {
    const combinedList = nums1.slice(0, m).concat(nums2.slice(0, n));
    combinedList.sort((a, b) => a - b);
    
    for (let i = 0; i < combinedList.length; i++) {
        nums1[i] = combinedList[i];
    }
    return nums1;
}

console.log(merge([1,2,3,0,0,0], 3, [2,5,6], 3));

File: LeetCode/88. Merge Sorted Array/merge.py
from typing import List


class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        combinedList = nums1[:m] + nums2[:n] 
        nums1[:] = sorted(combinedList)    
        return nums1
        
solution = Solution()
print(solution.merge([1,2,3,0,0,0], 3, [2,5,6], 3))